src/knightos/display.c

```diff
@@ -330,6 +330,50 @@ void draw_short(SCREEN *screen, unsigned char x, unsigned char y, unsigned short
 	screen; x; y; value;
 }
 
+void draw_signed(SCREEN* screen, unsigned char x, unsigned char y, signed short value) {
+    if(value < 0){
+        draw_char(screen, x, y, '-');
+        x += 4;
+        value = -value;
+    }
+    draw_short(screen, x, y, value);
+    return;
+}
+
+void draw_float(SCREEN* screen, unsigned char x, unsigned char y, float value){
+    /* Implementation is weird and slow because of non-working snprintf. Indeed, %f format strings are not yet implemented */
```

Comment:
@aviallon what was wrong with snprintf?

Reply:
%f doesn't work at all (it's ignored). I don't know if it is standard, but every libc I know supports it. And it really is useful.

Reply:
I can't remember why we didn't implement %f, and there's probably a good reason, but feel free to experiment with https://github.com/KnightOS/libc/blob/master/src/format.c

Reply:
Worst case, I'd like to see %f support using this function as a base.

Reply:
Actually, this function has a major issue : 
it does not support floats whose integer part is greater than `INT_MAX`...
In order to have full float support, I would have to go char by char... which may be even slower :disappointed:

Reply:
Let me know when this is ready for another review :)